instrumentation/github.com/gorilla/mux/otelmux/test/mux_test.go

```diff
@@ -23,6 +25,60 @@ import (
 	"go.opentelemetry.io/otel/trace"
 )
 
+type recordingResponseWriter struct {
+    writer     http.ResponseWriter
+    statusCode int
+    size       int
+}
+
+func (r *recordingResponseWriter) Header() http.Header {
+    return r.writer.Header()
+}
+
+func (r *recordingResponseWriter) Write(b []byte) (int, error) {
+    if r.statusCode == 0 {
+        r.statusCode = http.StatusOK
+    }
+    size, err := r.writer.Write(b)
+    r.size += size
+    return size, err
+}
+
+func (r *recordingResponseWriter) WriteHeader(statusCode int) {
+    r.statusCode = statusCode
+    r.writer.WriteHeader(statusCode)
+}
+
+func (r *recordingResponseWriter) Hijack() (net.Conn, *bufio.ReadWriter, error) {
+    hj, ok := r.writer.(http.Hijacker)
+    if !ok {
+        return nil, nil, http.ErrNotSupported
+    }
+    return hj.Hijack()
+}
+
+type mockHijacker struct {
+    http.ResponseWriter
+}
+
+func (m *mockHijacker) Hijack() (net.Conn, *bufio.ReadWriter, error) {
+    conn, rw := net.Pipe()
+    return conn, bufio.NewReadWriter(bufio.NewReader(rw), bufio.NewWriter(rw)), nil
+}
+
+func TestRecordingResponseWriterHijack(t *testing.T) {
+    mockWriter := &mockHijacker{httptest.NewRecorder()}
+    recWriter := &recordingResponseWriter{writer: mockWriter}
+
+    conn, rw, err := recWriter.Hijack()
+    require.NoError(t, err, "Hijack should not return an error")
+    assert.NotNil(t, conn, "Connection should not be nil")
+    assert.NotNil(t, rw, "ReadWriter should not be nil")
```

Comment:
```suggestion
    require.NoError(t, err)
    assert.NotNil(t, conn)
    assert.NotNil(t, rw)
```

Testify's own default message will be the same as this, so no need to set a custom one.